apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -534,4 +541,97 @@ export const pageRouter = router({
       }
       return permalinkTree
     }),
+
+  createIndexPage: protectedProcedure
+    .input(createIndexPageSchema)
+    .mutation(async ({ ctx, input: { siteId, parentId } }) => {
+      await validateUserPermissionsForResource({
+        userId: ctx.user.id,
+        siteId: siteId,
+        action: "create",
+        resourceId: parentId,
+      })
+
+      // Validate whether parentId exists and is a Folder or Collection
+      const parent = await db
+        .selectFrom("Resource")
+        .where("Resource.id", "=", parentId)
+        .where("Resource.siteId", "=", siteId)
+        .where("Resource.type", "in", [
+          ResourceType.Folder,
+          ResourceType.Collection,
+        ])
+        .select(["title", "permalink", "type"])
+        .executeTakeFirst()
+
+      if (!parent) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: "Parent resource not found or is not a folder/collection",
+        })
+      }
+
+      // Get the full permalink of the parent
+      const parentFullPermalink = await getResourceFullPermalink(
+        siteId,
+        parseInt(parentId),
+      )
+      if (!parentFullPermalink) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: "Could not determine parent's full permalink",
+        })
+      }
+
+      return db.transaction().execute(async (tx) => {
+        const blob = await tx
+          .insertInto("Blob")
+          .values({
+            content: jsonb({
+              page: {
+                title: parent.title,
+                permalink: parentFullPermalink,
```

Comment:
clarification: could you explain abit more why we need this? based off my understanding, we get the `r.permalink` from the `resource` and use that to construct the full permalink so i'm not sure if this is required/used

Reply:
@seaerchin current implementation mimicked that of oncall, but just clarified with ZJ that this is a legacy thing that's no longer needed - removed in https://github.com/opengovsg/isomer/pull/1065/commits/8fd3184d08efba2f82d850174ae72028be02e5de . thanks!